server/public/shared/httpservice/client.go

```diff
@@ -155,6 +159,26 @@ func dialContextFilter(dial DialContextFunction, allowHost func(host string) boo
 	}
 }
 
+func getProxyFn() func(r *http.Request) (*url.URL, error) {
+	proxyFromEnvFn := httpproxy.FromEnvironment().ProxyFunc()
+	return func(r *http.Request) (*url.URL, error) {
+		// Mitigating against MM-61938
+		// TODO: Remove this code once the above issue is fixed upstream.
+		if r.URL != nil {
+			if addr, err := netip.ParseAddr(r.URL.Hostname()); err == nil && addr.Is6() && addr.Zone() != "" {
+				// Strip IPv6 zone and rebuild URL
+				if r.URL.Port() != "" {
+					r.URL.Host = addr.WithZone("").String() + ":" + r.URL.Port()
+				} else {
+					r.URL.Host = addr.WithZone("").String()
+				}
```

Comment:
Might simplify to:

```go
portSuffix := ""
if r.URL.Port() != "" {
	portSuffix = ":" + r.URL.Port()
}
r.URL.Host = addr.WithZone("").String() + portSuffix
```

Reply:
point of view: still 5 lines + a new variable :p